src/test/scala/cromwell/util/SampleWdl.scala

```diff
@@ -1668,4 +1668,32 @@ object SampleWdl {
       """.stripMargin
     override val rawInputs = Map.empty[String, String]
   }
+
+  /**
+    * Outputs which reference themselves.
+    */
+  object InputsAndOutputsAutoReferentiation extends SampleWdl {
```

Comment:
This implies in some way that the ordering of inputs/outputs declaration matters. Is it a spec requirement ?
What about this for instance:
```
String a = "hello"
String a = "world"
```